x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/processes/processes.tsx

```diff
@@ -0,0 +1,179 @@
+  );
+
+  const [sortBy, setSortBy] = useState<SortBy>({
+    name: 'cpu',
+    isAscending: false,
+  });
+
+  const hostTerm = useMemo(() => {
+    const field = getFieldByType(nodeType) ?? nodeType;
+    return { [field]: node.name };
+  }, [node, nodeType]);
+
+  const {
+    loading,
+    error,
+    response,
+    makeRequest: reload,
+  } = useProcessList(hostTerm, currentTime, sortBy, parseSearchString(hostFlyoutOpen.searchFilter));
+
+  const debouncedSearchOnChange = useMemo(
+    () =>
+      debounce<(queryText: string) => void>(
+        (queryText) => setHostFlyoutOpen({ searchFilter: queryText }),
+        500
+      ),
+    [setHostFlyoutOpen]
+  );
+
+  const searchBarOnChange = useCallback(
+    ({ query, queryText }) => {
+      setSearchBarState(query);
+      debouncedSearchOnChange(queryText);
+    },
+    [setSearchBarState, debouncedSearchOnChange]
+  );
+
+  const clearSearchBar = useCallback(() => {
+    setSearchBarState(Query.MATCH_ALL);
+    setHostFlyoutOpen({ searchFilter: '' });
+  }, [setHostFlyoutOpen]);
+
+  return (
+    <TabContent>
+      <ProcessListContextProvider hostTerm={hostTerm} to={currentTime}>
+        <SummaryTable
+          isLoading={loading}
+          processSummary={(!error ? response?.summary : null) ?? { total: 0 }}
+        />
+        <EuiSpacer size="m" />
+        <EuiText>
+          <h4>
+            {i18n.translate('xpack.infra.metrics.nodeDetails.processesHeader', {
+              defaultMessage: 'Top processes',
+            })}{' '}
+            <EuiIconTip
+              aria-label={i18n.translate(
+                'xpack.infra.metrics.nodeDetails.processesHeader.tooltipLabel',
+                {
+                  defaultMessage: 'More info',
+                }
+              )}
+              size="m"
+              type="iInCircle"
+              content={i18n.translate(
+                'xpack.infra.metrics.nodeDetails.processesHeader.tooltipBody',
+                {
+                  defaultMessage:
+                    'The table below aggregates the top CPU and top memory consuming processes. It does not display all processes.',
+                }
+              )}
+            />
+          </h4>
+        </EuiText>
+        <EuiSpacer size="m" />
+        <EuiSearchBar
+          query={searchBarState}
+          onChange={searchBarOnChange}
+          box={{
+            incremental: true,
+            placeholder: i18n.translate('xpack.infra.metrics.nodeDetails.searchForProcesses', {
+              defaultMessage: 'Search for processes…',
+            }),
+          }}
+          filters={[
+            {
+              type: 'field_value_selection',
+              field: 'state',
+              name: 'State',
+              operator: 'exact',
+              multiSelect: false,
+              options: Object.entries(STATE_NAMES).map(([value, view]: [string, string]) => ({
+                value,
+                view,
+              })),
```

Comment:
**nit**: this is a static value that has no dependencies from the component lifecycle, you can map it at a module level once and pass it here to avoid recreating this on each render.

Reply:
yes, you are right, I moved that outside 👍